src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,237 @@
+package org.openstreetmap.atlas.geography.atlas.builder.proto;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.Latitude;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Longitude;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.AtlasMetaData;
+import org.openstreetmap.atlas.geography.atlas.builder.AtlasSize;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlasBuilder;
+import org.openstreetmap.atlas.proto.ProtoAtlasContainer;
+import org.openstreetmap.atlas.proto.ProtoLine;
+import org.openstreetmap.atlas.proto.ProtoLocation;
+import org.openstreetmap.atlas.proto.ProtoPoint;
+import org.openstreetmap.atlas.proto.ProtoTag;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.atlas.streaming.resource.WritableResource;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Build an Atlas from a naive proto serialized file, or write an Atlas to a naive proto serialized
+ * file. At this stage, the builder can only handle Atlases that have tagged Points and Lines.
+ *
+ * @author lcram
+ */
+public class ProtoAtlasBuilder
+{
+    private static final Logger logger = LoggerFactory.getLogger(ProtoAtlasBuilder.class);
+
+    private static final String TAG_DELIMITER = "=";
+
+    /**
+     * Read a resource in naive ProtoAtlas format into a PackedAtlas.
+     *
+     * @param resource
+     *            the resource in naive ProtoAtlas format
+     * @return the constructed PackedAtlas
+     */
+    public PackedAtlas read(final Resource resource)
+    {
+        ProtoAtlasContainer protoAtlasContainer = null;
+
+        // First, we need to construct the container object from the proto binary
+        try
+        {
+            protoAtlasContainer = ProtoAtlasContainer.parseFrom(resource.readBytesAndClose());
+        }
+        // TODO do something smarter here?
```

Comment:
Only add the caught exception to the CoreException argument list so the stack trace isn't lost.

throw new CoreException("Error parsing the protobuf", exception);